modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/DDMObjectValidationRuleEngineImpl.java

```diff
@@ -46,14 +47,22 @@ public Map<String, Object> execute(
 			"invalidScript", false
 		).build();
 
+		Map<String, Object> oldVariables =
+			(Map<String, Object>)inputObjects.get("originalObjectEntry");
+
+		Map<String, Object> variables = (Map<String, Object>)inputObjects.get(
+			"objectEntry");
+
 		try {
 			DDMExpression<Boolean> ddmExpression =
 				_ddmExpressionFactory.createExpression(
 					CreateExpressionRequest.Builder.newBuilder(
 						script
+					).withDDMExpressionParameterAccessor(
+						new ObjectDDMExpressionParameterAccessor(oldVariables)
 					).build());
 
-			ddmExpression.setVariables(inputObjects);
+			ddmExpression.setVariables(variables);
```

Comment:
@p-albuquerque I feel like we are changing the current behavior here.. We were passing `inputObjects` but now we are only passing the `objectEntry` from the `inputObjects`.. It doesn't seem safe